futures-core/src/async_trait/mod.rs

```diff
@@ -0,0 +1,210 @@
+//! Asynchronous values.
+
+use core::mem::Pin;
+use core::marker::Unpin;
+
+use {Poll, PollResult, task};
+
+#[cfg(feature = "either")]
+mod either;
+
+/// An `Async` value represents an asychronous computation.
+///
+/// An `Asyn` is a value that may not have finished computing yet. This kind of
+/// "asynchronous value" makes it possible for a thread to continue doing useful
+/// work while it waits for the value to become available.
+///
+/// The ergonomics and implementation of the `Async` trait are very similar to
+/// the `Iterator` trait in that there is just one method you need to
+/// implement, but you get a whole lot of others for free as a result. These
+/// other methods allow you to chain together large computations based on
+/// async values, which will automatically handle asynchrony for you.
+///
+/// # The `poll` method
+///
+/// The core method of `Async`, `poll`, *attempts* to resolve the computation into a
+/// final value. This method does not block if the value is not ready. Instead,
+/// the current task is scheduled to be woken up when it's possible to make
+/// further progress by `poll`ing again. The wake up is performed using
+/// `cx.waker()`, a handle for waking up the current task.
+///
+/// When using an async value, you generally won't call `poll` directly, but instead
+/// use combinators to build up asynchronous computations. A complete
+/// computation can then be spawned onto an
+/// [executor](../futures_core/executor/trait.Executor.html) as a new, independent
+/// task that will automatically be `poll`ed to completion.
+///
+/// # Combinators
```

Comment:
Here and above you mention combinators on `Async`-- this should link to `AsyncExt` (when it exists).